pubnub-chat-impl/src/commonTest/kotlin/com/pubnub/integration/ChatIntegrationTest.kt

```diff
@@ -738,10 +767,143 @@ class ChatIntegrationTest : BaseChatIntegrationTest() {
+        assertEquals(
+            chat.currentUser.asImpl().copy(updated = null, lastActiveTimestamp = null),
+            secondResult.hostMembership.user.asImpl().copy(updated = null, lastActiveTimestamp = null)
+        )
+        assertEquals(someUser, secondResult.inviteeMembership.user.asImpl())
+        assertEquals(secondResult.channel, secondResult.hostMembership.channel)
+        assertEquals(secondResult.channel, secondResult.inviteeMembership.channel)
+
+        // verify no duplicate channels created - fetch channel directly
+        val fetchedChannel = chat.getChannel(firstChannelId).await()
+        assertNotNull(fetchedChannel, "Original channel should still exist")
+        assertEquals(firstChannelId, fetchedChannel.id)
+    }
+
+    @Test
+    fun isPresent_withStateTransitions_shouldReflectCorrectState() = runTest {
+        // given - create a test channel
+        val testChannelId = randomString()
+        val testChannel = chat.createChannel(testChannelId).await()
+
+        // initially, current user should not be present (not subscribed)
+        val initiallyPresent = chat.isPresent(chat.currentUser.id, testChannelId).await()
+        assertFalse(initiallyPresent, "User should not be present before subscribing")
+
+        // when - user subscribes to the channel
+        pubnub.test(backgroundScope, checkAllEvents = false) {
+            var subscription: AutoCloseable? = null
+            pubnub.awaitSubscribe(listOf(testChannelId)) {
+                subscription = testChannel.connect { }
+            }
+
+            // wait for presence to update
+            delayInMillis(2000)
+
+            // then - user should be present
+            val nowPresent = chat.isPresent(chat.currentUser.id, testChannelId).await()
+            assertTrue(nowPresent, "User should be present after subscribing")
+
+            // when - user unsubscribes
+            subscription?.close()
+
+            // wait for presence to update
+            delayInMillis(2000)
+
+            // then - user should not be present anymore
+            val afterUnsubscribe = chat.isPresent(chat.currentUser.id, testChannelId).await()
+            assertFalse(afterUnsubscribe, "User should not be present after unsubscribing")
+        }
+    }
+
     private suspend fun assertPushChannels(chat: Chat, expectedNumberOfChannels: Int) {
         val pushChannels = chat.getPushChannels().await()
         assertEquals(expectedNumberOfChannels, pushChannels.size)
     }
+
+    @Test
+    fun wherePresent_whenUserNotSubscribed_shouldReturnEmptyList() = runTest {
+        // given - a user not subscribed to any channels
+        val channels = chat.wherePresent(chat.currentUser.id).await()
+
+        // then - should return empty list
+        assertTrue(channels.isEmpty(), "User not subscribed to any channel should have empty wherePresent result")
+    }
+
+    @Test
+    fun wherePresent_whenUserSubscribedToChannel_shouldReturnChannelId() = runTest {
```

Comment:
What's the difference between these tests?

- `wherePresent_whenUserSubscribedToChannel_shouldReturnChannelId`
- `wherePresent_whenUserSubscribedToMultipleChannels_shouldReturnAllChannelIds`? 
 
Is the second test covering what's tested in the first one?
If yes, I would keep `wherePresent_whenUserSubscribedToMultipleChannels_shouldReturnAllChannelIds` only